src/java.base/share/classes/java/lang/ProcessHandleImpl.java

```diff
@@ -519,8 +519,7 @@ public boolean equals(Object obj) {
         if (this == obj) {
             return true;
         }
-        if (obj instanceof ProcessHandleImpl) {
-            ProcessHandleImpl other = (ProcessHandleImpl) obj;
+        if (obj instanceof ProcessHandleImpl other) {
             return (pid == other.pid) &&
                     (startTime == other.startTime
                         || startTime == 0
```

Comment:
This one can be a single expression. Otherwise, looks good.

Reply:
Is the short-circuiting using && and || a problem? In this case it does not save much time over fully computing the value.

Reply:
Well spotted, Chris. Fixed in commit f7924d2